### PR TITLE
gitlab: Refresh pipeline helper

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

Adds retries to gitlab pipeline.refresh() as @amenasria mentioned

### Additional Notes

One retry example:
image

### CODE REVIEW

File: tasks/libs/ciproviders/gitlab_api.py [hunk 1]
```diff
@@ -72,6 +76,36 @@ def get_gitlab_repo(repo='DataDog/datadog-agent', token=None) -> Project:
     return repo
 
 
+def refresh_pipeline(pipeline: ProjectPipeline, max_retries=2, retry_delay=1):
```
Comment (FlorentClarret) [inline tasks/libs/ciproviders/gitlab_api.py:79]: I think using a decorator (either custom or with something like [tenacity](https://tenacity.readthedocs.io/en/latest/)) on the function you want to retry would be more generic and easier to maintain. What do you think?

Comment (amenasria) [reply tasks/libs/ciproviders/gitlab_api.py:79]: +1000 on this !

Comment (amenasria) [reply tasks/libs/ciproviders/gitlab_api.py:79]: + the traceback logic you started to implement is really easily handled in the __exit__ function of a context manager

**Review by amenasria — APPROVED** [review]
Left a few comments, otherwise LGTM !


**👏 praise:** ‏Thanks for this !!

**Pull request merged**